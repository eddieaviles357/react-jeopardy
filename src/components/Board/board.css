.Board {
  flex-direction: row;
  align-items: normal;
  display: flex;
  height: 100vh;
  width: 100vw;
  overflow: auto;
}

.category-header {
  padding: 5px 0;
  font-size: 20px;
  text-transform: capitalize;
  text-align: center;
  color: #441fff;;

  background-color: #ffff00;
  /* border: 2px solid #ffffff; */
}

.Board-column {
  display: flex;
  height: 100%;
  flex-direction: column;
}

.Board-clue {
  position: relative;
  display: flex;
  align-items: center;
  justify-content: center;

  padding: 10px;  

  min-height: 100px;
  min-width: 200px;
  height: calc(100vh / 5);
  width: calc(100vw / 6);
  
  color: #fff;

  font-size: 2rem;

  background-color: #441fff;
  border: 1px solid #aaa;
}

.Active {
  display: block;
  position: fixed;
  top: 50%;
  left: 50%; 
  width: 100vw;
  height: 100vh;

  font-size: 30px;
  padding: 10px;

  transform: translate(-50%, -50%);

  border: 3px solid #ffff00;
  /* border-radius: 10px; */
  background-color: #441fff;
  z-index: 100;

  overflow: auto;
}

.clue-choices-container {
  padding: 10px;
  display: flex;
  align-items: flex-end;
  justify-content: space-between;

  border-radius: 10px;
}

#go-back,
#reveal-answer {
  padding: 10px;

  cursor: pointer;

  border-radius: 10px;
  border: 1px solid #000000;

  background-color: #ffff00;
  color: #0000ff;
}

.Active-qa-container,
.Active-revealed {
  display: block;
  text-align: center;
}

.Active-qa-container {
  color: white;
  padding: 15px 0 0 0;
}

.Active-revealed {
  font-size: 3.5rem;
  color: yellow;
}

.Active hr {
  margin: 20px 0;
  height: 3px;

  background-color: #ffff00;

  border: 2px solid #ffff00;
  border-radius: 10px;

}

.answered {
  color: gray;
}