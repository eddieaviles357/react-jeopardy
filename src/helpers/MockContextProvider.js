"use strict";

import { useState } from 'react'
import JeopardyContext from '../JeopardyContext'

const MockContextProvider = ({ children }) => {
    const [jpardyData, setJpardyData] = useState([])
    const [isPlaying, setIsPlaying] = useState(false)
    const [isLoading, setIsLoading] = useState(false)
    // used to avoid multiple calls to api
    const [isPlayedBtnStarted, setisPlayedBtnStarted] = useState(false)

    const startGame = async () => {
        try {
            if (isPlayedBtnStarted) return
            setisPlayedBtnStarted(true)
            setIsLoading(true)
            setJpardyData(await setBoardData())
            setIsPlaying(true)
            setIsLoading(false)
        } catch (err) {
            return <div>Sorry something went wrong.Please Refresh</div>
        }
    }
    return (
        <JeopardyContext.Provider value={{
            jpardyData,
            setJpardyData,
            isPlaying,
            setIsPlaying,
            startGame,
            isLoading,
        }}>
            {children}
        </JeopardyContext.Provider>
    )
}

export default MockContextProvider;