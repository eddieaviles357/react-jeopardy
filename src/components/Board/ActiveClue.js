import React from 'react'
import './board.css'

const ActiveClue = ({ content, answer }) => {
  
  const activeClueClickHandler = (evt) => {
    // avoids parents click events to trigger
    evt.stopPropagation()
    const id = evt.target.id
    if(id === 'go-back') { 
      console.log('goback')
    }
    if(id === 'reveal-answer') {
      console.log('reveal-answer')
    }
  }

  const revealAnswer = (evt) => {
    evt.stopPropagation()
    console.log('answer', answer)
  }

  return (
    <div className='active' onClick={activeClueClickHandler}>
      <div className='clue-choice-container'>
        <span id='go-back'>Go Back</span>
        <span id='reveal-answer'>Reveal Answer</span>
      </div>
      { content }
    </div>
  )
}

export default ActiveClue
