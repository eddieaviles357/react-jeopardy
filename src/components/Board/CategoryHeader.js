import React from 'react'
import './board.css'

const CategoryHeader = ({ children }) => {
  return (
    <div className='category-header'>{children}</div>
  )
}

export default CategoryHeader
