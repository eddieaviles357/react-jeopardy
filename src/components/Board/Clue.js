import React, {useState} from 'react'
import './board.css'
import ActiveClue from './ActiveClue'

const Clue = ({ id, question, answer, value, showing }) => {
  const [isShowing, setIsShowing] = useState(showing)
  
  
  const expandClue = (evt) => {
    (isShowing === null) ? setIsShowing( true) : setIsShowing( false )
    const id = evt.target.id
    // extract index from clue
    const categoryIdx = +id[0]
    const clueIdx = +id[id.length - 1]
    console.log(categoryIdx, clueIdx)
  }

  
  const content = (isShowing === null) ? value : (isShowing === true) ? question : answer

  return (
    <div 
      onClick={ expandClue }
      className='Board-clue' 
      id={id}
      >
      { content }
      { isShowing && <ActiveClue content={content} answer={answer} setIsShowing={setIsShowing}/> }
      </div>
  )
}

export default Clue
