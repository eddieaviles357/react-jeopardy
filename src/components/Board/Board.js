import React, { useContext } from 'react'
import JeopardyContext from '../../JeopardyContext'
import CategoryHeader from './CategoryHeader'
import Clue from './Clue'
import { v4 as uuidv4 } from 'uuid';
import { DEFAULT_CLUE } from '../../constants'

const Board = () => {
  const { jpardyData } = useContext( JeopardyContext )
  
  // question, answer, showing
  const jpardyClues = ( cluesArr, idx1 ) => {
    // if api returns an empty array of clues use default values
    while(cluesArr.length < 5) {
      cluesArr.push( DEFAULT_CLUE )
    }

    return (
    cluesArr.map( ({ question,answer,value,showing }, idx2) => ( 
      <Clue 
        key={ uuidv4()} 
        id={ `${idx1} - ${idx2}` }
        question={ question }
        answer={ answer }
        value={ value }
        showing={ showing }
        /> 
    ))
  )}

  console.log('Board.Component', jpardyData)

  return (
    <div className='Board'>
    {jpardyData.map( ({ title, clues }, idx1 ) => ( 
        <div className='Board-column' key={ uuidv4() }>
          <CategoryHeader>{ title }</CategoryHeader> 
          { jpardyClues(clues, idx1) }
        </div>
      ))}
    </div>
  )
}

export default Board