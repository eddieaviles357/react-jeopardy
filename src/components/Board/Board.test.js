"use strict";

import { render } from '@testing-library/react'
import Board from './Board'
import MockContextProvider from '../../helpers/MockContextProvider';

test('Does not crash', () => {
    render(
        <MockContextProvider>
            <Board />
        </MockContextProvider>);
});

test('Matches snapshot', () => {
    const board = render(
        <MockContextProvider>
            <Board />
        </MockContextProvider>
    );
    expect(board.asFragment()).toMatchSnapshot();
})