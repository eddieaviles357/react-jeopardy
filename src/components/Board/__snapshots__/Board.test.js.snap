// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Board"
  >
    <div
      class="Board-column"
    >
      <div
        class="category-header"
      >
        Test title
      </div>
      <div
        class="Board-clue "
        id="0 - 0"
      >
        100
      </div>
      <div
        class="Board-clue "
        id="0 - 1"
      >
        100
      </div>
      <div
        class="Board-clue "
        id="0 - 2"
      >
        100
      </div>
      <div
        class="Board-clue "
        id="0 - 3"
      >
        100
      </div>
      <div
        class="Board-clue "
        id="0 - 4"
      >
        100
      </div>
    </div>
  </div>
</DocumentFragment>
`;
